plugins/services/src/js/stores/MarathonStore.js

```diff
@@ -470,7 +471,12 @@ class MarathonStore extends GetSetBaseStore {
     data.items.forEach(item => {
       if (item.items && Array.isArray(item.items)) {
         this.injectGroupsWithPackageImages(item);
-      } else if (ServiceValidatorUtil.isFrameworkResponse(item)) {
+      } else if (
+        (ServiceValidatorUtil.isFrameworkResponse(item) ||
+          ServiceValidatorUtil.isApplicationResponse(item)) &&
```

Comment:
I wonder if we should add `Pod` as well. So basically every item that has `DCOS_PACKAGE_NAME`

Reply:
I don't know if Pods can have images. If everything can, we can just check for name and images.
But I am not sure if this change will be in scope here. The bug is that some catalog services don't have icons.

Reply:
Ok, creeping scope is a fair observation and the bug is about service coming from the catalog/universe. I believe one can create a catalog package that will be essentially a Pod.